recipes/tacker_server.rb

```diff
@@ -123,9 +123,10 @@ class ::Chef::Recipe
   action :nothing
 end
 
+flavor_key = 'resource_types:OS::Nova::Flavor'
 execute 'Allow users in non-admin projects with admin roles to create flavors.' do
-  command 'sudo sed -i.bak \'s/"resource_types:OS::Nova::Flavor.*/"resource_types:OS::Nova::Flavor": "role:admin",/\' /etc/heat/policy.json'
-  action :nothing
+  command "sudo sed -i.bak 's|\"#{flavor_key}.*|\"#{flavor_key}\": \"role:admin\",|' /etc/heat/policy.json"
+  not_if "grep '#{flavor_key}.*role:admin' " '/etc/heat/policy.json'
```

Comment:
this should be clearly marked as an workaround since config files should usually not be edited with sed

Reply:
Done.